src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java

```diff
@@ -281,32 +274,42 @@ private Optional<NetworkModificationResult> saveAndApplyModifications(UUID targe
         return Optional.empty();
     }
 
-    @Transactional
     public Optional<NetworkModificationResult> duplicateModifications(UUID targetGroupUuid,
                                                                       UUID networkUuid, String variantId,
                                                                       ReportInfos reportInfos, List<UUID> modificationsUuids) {
-        List<ModificationEntity> modificationsEntities = networkModificationRepository.getModificationsEntities(modificationsUuids);
-        List<ModificationEntity> duplicatedModificationsEntities = modificationsEntities.stream().map(m -> m.toModificationInfos().toEntity()).collect(Collectors.toList());
-        return saveAndApplyModifications(targetGroupUuid, networkUuid, variantId, reportInfos, duplicatedModificationsEntities);
+        List<ModificationInfos> modificationInfos = networkModificationRepository.getModificationsInfos(modificationsUuids);
+        saveModifications(targetGroupUuid, modificationInfos);
+        return applyModifications(networkUuid, variantId, reportInfos, modificationInfos);
     }
 
     public UUID createModificationInGroup(@NonNull ModificationInfos modificationsInfos) {
         UUID groupUuid = UUID.randomUUID();
-        networkModificationRepository.saveModifications(groupUuid, List.of(modificationsInfos.toEntity()));
+        saveModifications(groupUuid, modificationsInfos);
         return groupUuid;
     }
 
-    @Transactional
-    public Optional<NetworkModificationResult> duplicateModificationsInGroup(UUID targetGroupUuid,
-                                                                             UUID networkUuid, String variantId,
-                                                                             ReportInfos reportInfos,
-                                                                             UUID originGroupUuid) {
-        List<ModificationEntity> duplicatedModificationsEntities = networkModificationRepository.copyModificationsEntities(originGroupUuid);
-        return saveAndApplyModifications(targetGroupUuid, networkUuid, variantId, reportInfos, duplicatedModificationsEntities);
+    public Optional<NetworkModificationResult> duplicateModifications(UUID targetGroupUuid,
+                                                                      UUID networkUuid, String variantId,
+                                                                      ReportInfos reportInfos,
+                                                                      UUID originGroupUuid) {
+        List<ModificationInfos> modificationInfos = networkModificationRepository.getModificationsInfos(originGroupUuid, false);
+        saveModifications(targetGroupUuid, modificationInfos);
+        return applyModifications(networkUuid, variantId, reportInfos, modificationInfos);
     }
 
```

Comment:
@Transactional is removed, do we need rollback if a runtime exception occurs while invoking applyModifications()???